src/Microsoft.DotNet.Darc/src/Darc/Operations/AddBuildToChannelOperation.cs

```diff
@@ -136,10 +136,36 @@ private async Task<int> PromoteBuildAsync(Build build, Channel targetChannel, IR
                 queueTimeVariables)
                 .ConfigureAwait(false);
 
-            Console.WriteLine($"Build {build.Id} will be assigned to channel '{targetChannel.Name}' once this promotion build finishes: " +
-                $"https://{BuildPromotionPipelineAccountName}.visualstudio.com/{BuildPromotionPipelineProjectName}/_build/results?buildId={azdoBuildId}&view=results");
+            var promotionBuildUrl = $"https://{BuildPromotionPipelineAccountName}.visualstudio.com/{BuildPromotionPipelineProjectName}/_build/results?buildId={azdoBuildId}";
 
-            return Constants.SuccessCode;
+            Console.WriteLine($"Build {build.Id} will be assigned to channel '{targetChannel.Name}' once this promotion build finishes: {promotionBuildUrl}");
+
+            if (_options.NoWaitPromotionBuild)
+            {
+                Console.WriteLine("Returning before the promotion build finish.");
+                return Constants.SuccessCode;
+            }
+
+            var waitIntervalInSeconds = TimeSpan.FromSeconds(60);
+            AzureDevOpsBuild promotionBuild;
+
+            do
+            {
+                Console.WriteLine($"Waiting more {waitIntervalInSeconds.TotalSeconds} seconds for promotion build to complete.");
+                await Task.Delay(waitIntervalInSeconds);
+                promotionBuild = await azdoClient.GetBuildAsync(BuildPromotionPipelineAccountName, BuildPromotionPipelineProjectName, azdoBuildId);
```

Comment:
What happens if this call get the build fails? We should probably let the user know that we can't keep the track of the build status anymore, but that their assets will be published once the build we provided for them earlier is successful.

Reply:
Makes sense. Going up in next push.